src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioSymbolNavigationService.cs

```diff
@@ -182,6 +183,17 @@ public bool TryNavigateToSymbol(ISymbol symbol, Project project, OptionSet optio
             return true;
         }
 
+        public async Task<MetadataAsSourceFile?> GetAndOpenGeneratedFileAsync(ISymbol symbol, Project project, CancellationToken cancellationToken)
+        {
+            var result = await _metadataAsSourceFileService.GetGeneratedFileAsync(
+                project, symbol, allowDecompilation: false, cancellationToken).ConfigureAwait(false);
+
+            var openDocumentService = IServiceProviderExtensions.GetService<SVsUIShellOpenDocument, IVsUIShellOpenDocument>(_serviceProvider);
```

Comment:
If feels a little odd that to have a document be seen by the LSP server, we have to call a VS service to open it, and just avoid showing it. How necessary is that?

I guess what I'm wondering is, is this just working because we call VS, which ends up calling into us, and is there a way to circumvent that and call into our own service to open the document? Or is the VS bit actually 100% necessary?

Reply:
+1 to this is there any way to sidestep this?  

This may also require the UI thread which we don't want to be doing if at all possible in LSP

Reply:
> is there a way to circumvent that and call into our own service to open the document?

This is what I'm trying to figure out, but I haven't been able to find an alternative so far. Calling `OpenDocument()` on the metadata workspace doesn't work since the metadata workspace apparently doesn't support opening documents.

Reply:
Resolving convo - per offline conversation, going through VS has been noted to be likely OK for the near-term.
I've extracted the new interface method into a new service to hopefully keep things cleaner.